internal/pkg/source/source.go

```diff
@@ -119,73 +119,59 @@ func sourcesFromBlocks(fn *ssa.Function, conf *config.Config, taggedFields field
-			default:
-				continue
-
-			// Values produced by sanitizers are not sources.
-			case *ssa.Alloc:
-				if isProducedBySanitizer(v, conf) {
-					continue
-				}
-
-			// Values produced by sanitizers are not sources.
-			// Values produced by field propagators are.
-			case *ssa.Call:
-				if isProducedBySanitizer(v, conf) {
-					continue
-				}
-
-				if propagators.IsFieldPropagator(v) {
-					sources = append(sources, New(v))
-					continue
-				}
-
-			// A panicky type assert like s := e.(*core.Source) does not result in ssa.Extract
-			// so we need to create a source if the type assert is panicky i.e. CommaOk is false
-			// and the type being asserted is a source type.
-			case *ssa.TypeAssert:
-				if !v.CommaOk && IsSourceType(conf, taggedFields, v.AssertedType) {
-					sources = append(sources, New(v))
-				}
-
-			// An Extract is used to obtain a value from an instruction that returns multiple values.
-			// If the extracted value is a Pointer to a Source, it won't have an Alloc, so we need to
-			// identify the Source from the Extract.
-			case *ssa.Extract:
-				t := v.Tuple.Type().(*types.Tuple).At(v.Index).Type()
-				if _, ok := t.(*types.Pointer); ok && IsSourceType(conf, taggedFields, t) {
-					sources = append(sources, New(v))
-				}
-				continue
-
-			// value received from chan
-			case *ssa.UnOp:
-				// not a <-chan operation
-				if v.Op != token.ARROW {
-					continue
-				}
-
-			// value obtained through a field or an index operation
-			case *ssa.Field, *ssa.FieldAddr, *ssa.Index, *ssa.IndexAddr, *ssa.Lookup:
-
-			// chan or map value (arrays and slices are created using Allocs)
-			case *ssa.MakeMap, *ssa.MakeChan:
-			}
-
-			// all of the instructions that the switch lets through are values as per ssa/doc.go
-			if v := instr.(ssa.Value); IsSourceType(conf, taggedFields, v.Type()) {
-				sources = append(sources, New(v.(ssa.Node)))
+			if n := instr.(ssa.Node); isSourceNode(n, conf, propagators, taggedFields) {
+				sources = append(sources, New(n))
 			}
 		}
 	}
 	return sources
 }
 
+func isSourceNode(n ssa.Node, conf *config.Config, propagators fieldpropagator.ResultType, taggedFields fieldtags.ResultType) bool {
+	switch v := n.(type) {
+	// All sources are explicitly identified.
+	default:
+		return false
+
+	// Values produced by sanitizers are not sources.
+	case *ssa.Alloc:
+		return !isProducedBySanitizer(v, conf) && IsSourceType(conf, taggedFields, n.(ssa.Value).Type())
+
+	// Values produced by sanitizers are not sources.
+	// Values produced by field propagators are.
+	case *ssa.Call:
+		return !isProducedBySanitizer(v, conf) &&
+			(propagators.IsFieldPropagator(v) || IsSourceType(conf, taggedFields, n.(ssa.Value).Type()))
+
+	// A type assertion can assert that an interface is of a source type.
+	// Only panicky type asserts will refer to the source Value.
+	// _, ok type assertions are checked as a source type in the ssa.Extract instruction.
```

Comment:
Nit: ... in the *case for the* ssa.Extract ...

Reply:
`case` specified.  Also removed how that case is doing its check - that belongs to that case's comment.